### ISSUE TITLE
Disable Zookeeper Admin Server?

### ISSUE THREAD

**Issue by christopher-cudennec** [salesforce/kafka-junit#49]
Hello Salesforce team!

First of all thanks for your great testing library. We use it a lot in our [open source library](https://github.com/SDA-SE/sda-dropwizard-commons/) to write our integration tests!

We just wanted to upgrade our Kafka libraries (from 2.2.x to 2.6.x) and ran into an issue with Zookeeper: Zookeeper also starts the admin server. Unfortunately it always uses port 8080 when starting up. Since we want to do parallel tests we frequently have binding issues.

I could solve the problem locally by setting the system property "zookeeper.admin.enableServer" to "false" in my build.gradle.

My question is: 
Do you think it might be an option to disable the admin server by default? I'm not sure if you would usually need it if you want to test Kafka. It would probably also speed up the startup times because there would be one resource less that needs to be started.
I could also try to provide a PR for that change if you agree.

Follow-up question:
Do you have any experience with the latest Kafka version, i.e. 2.7.x? You latest release only states the you validated against 2.5.x and 2.6.x.

Thanks a lot!

Christopher

**Comment by Crim** [comment]
Hi Christopher,

I think having the admin server disabled by default would be totally reasonable.  I suspect no one using this library makes use of it, but It may be worth adding it as a configurable option to re-enable it in [AbstractZookeeperTestResource](https://github.com/salesforce/kafka-junit/blob/9675c18b5221a156821d534129b9f03e6b6779b3/kafka-junit-core/src/main/java/com/salesforce/kafka/test/AbstractZookeeperTestResource.java#L31) similar to how we have various configurable properties in AbstractKafkaTestResource.

Definitely welcome a PullRequest :)


In regards to Kafka 2.7.x, I haven't personally tested it yet, but I suspect that it works without any problems.  Typically what I do to validate a new version is to just update my POM locally and run the test suite and/or  update the [travis.yml](https://github.com/salesforce/kafka-junit/blob/master/.travis.yml#L11) file with a new test build using the latest version and see if all the tests pass.  If so then I just update the README, a pretty relaxed processed.  

With TravisCI no longer being so generous with their free CI pipelines, I may need to upgrade the project to use another service.  Feel free to give it a go locally and let me know if it works, or if you run into troubles.

Thanks!

**Comment by christopher-cudennec** [comment]
Alright! I'll prepare a PR in the next couple of days and come back to you! Thanks for your feedback!

**Comment by christopher-cudennec** [comment]
Hi @Crim!

Some quick feedback for providing a PR: Some tests always fail locally because there are some resources missing. `KafkaTestClusterTest` and `KafkaTestServerTest` fail because they can't find `kafka.keystore.jks` + `kafka.truststore.jks`. I probably need to disable the tests.

Cheers, 

Christopher

**Comment by christopher-cudennec** [comment]
Ah, I just came across your travis file and found the scripts to generate the certificates. :)

**Comment by christopher-cudennec** [comment]
Hey @Crim !

I played around with your code and dug a bit into curator. My first problem was that the Admin Server was not started because the Jetty classes were not found in the classpath. The startup fails with an error message and a stack trace.

Because of that I tried updating Curator to the latest version. Then I wondered why the Admin Server was not started anymore. Long story short: The behaviour seems to have changed with the latest version. I even can't reproduce the error in our open source library sda-dropwizard-commons when I upgrade to Curator 5.1.0. What do you think about updating Curator as a fix for my problem?

Cheers,

Christopher

**Comment by Crim** [comment]
Hey Christopher,

Are you able to share a branch with your code changes so I can better understand?  

Is this similar to what you did #50 ?

I believe that upgrades zookeeper from ~3.4 to 3.6.  I'd have to check the release notes between versions, but perhaps they've either removed the admin port, or switched it to disabled by default?

**Comment by christopher-cudennec** [comment]
Hi!

It looks like Curator does not start the Admin Server anymore. The default behaviour of Zookeeper seems unchanged.
I think the reason is that the following class (and no other Curator class) does not call the AdminServerFactory:
[TestingZooKeeperMain](https://github.com/apache/curator/blob/master/curator-test/src/main/java/org/apache/curator/test/TestingZooKeeperMain.java)

Unfortunately I couldn't find an explanation or commit message from Curator if that was done intentionally.

In the end the only change that I did was updating the version in your pom.xml just like #50.

Cheers,

Christopher

**Comment by Crim** [comment]
Nice, that might be the best/easiest solution then.  I'm trying to consider any other side effects upgrading this may have, not sure I can come up with anything concrete, unless someone is depending on the older zookeeper/curator version explicitly....

**Comment by gquintana** [comment]
Same issue here when running multiple builds in parallel on the same Jenkins Slave:
```
2022-02-23 11:17:16.952 [Thread-2] ERROR o.a.c.test.TestingZooKeeperServer - From testing server (random state: false) for instance: InstanceSpec{dataDirectory=/tmp/kafka-unit6455329329378383964, port=33772, electionPort=40581, quorumPort=36054, deleteDataDirectoryOnClose=false, serverId=1, tickTime=-1, maxClientCnxns=1000, customProperties={}, hostname=127.0.0.1} org.apache.curator.test.InstanceSpec@59c43969
org.apache.zookeeper.server.admin.AdminServer$AdminServerException: Problem starting AdminServer on address 0.0.0.0, port 8080 and command URL /commands
	at org.apache.zookeeper.server.admin.JettyAdminServer.start(JettyAdminServer.java:189)
	at org.apache.zookeeper.server.ZooKeeperServerMain.runFromConfig(ZooKeeperServerMain.java:153)
	at org.apache.curator.test.TestingZooKeeperMain.runFromConfig(TestingZooKeeperMain.java:93)
	at org.apache.curator.test.TestingZooKeeperServer$1.run(TestingZooKeeperServer.java:148)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Failed to bind to /0.0.0.0:8080
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349)
	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310)
	at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
	at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.eclipse.jetty.server.Server.doStart(Server.java:401)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.zookeeper.server.admin.JettyAdminServer.start(JettyAdminServer.java:180)
	... 4 common frames omitted
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:461)
	at java.base/sun.nio.ch.Net.bind(Net.java:453)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:344)
	... 11 common frames omitted
```

**Comment by Crim** [comment]
I've just published version [v3.2.4](https://github.com/salesforce/kafka-junit/releases/tag/v3.2.4) and released to maven central.

If you continue to have issues with this in the new version please reopen this issue and we can continue discussions.  

Thanks again @gquintana for contributing the fix for this!
